docs/components/expression-metadata.js

```diff
@@ -5,99 +5,92 @@ import CompoundExpression from '../../src/style-spec/expression/compound_express
+}, {
+    type: 'boolean',
+    parameters: ['string', 'value']
+}, {
+    type: 'boolean',
+    parameters: ['string', 'value', 'collator']
+}, {
+    type: 'boolean',
+    parameters: ['number', 'value']
+}, {
+    type: 'boolean',
+    parameters: ['value', 'string']
+}, {
+    type: 'boolean',
+    parameters: ['value', 'string', 'collator']
+}, {
+    type: 'boolean',
+    parameters: ['value', 'number']
+}];
+
 const types = {
-    '==': [{
-        type: 'boolean',
-        parameters: ['string', 'string']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'string', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['number', 'number']
-    }, {
-        type: 'boolean',
-        parameters: ['boolean', 'boolean']
-    }, {
-        type: 'boolean',
-        parameters: ['null', 'null']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'value', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['number', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['boolean', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['null', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'string']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'string', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'number']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'boolean']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'null']
-    }],
-    '!=': [{
-        type: 'boolean',
-        parameters: ['string', 'string']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'string', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['number', 'number']
-    }, {
-        type: 'boolean',
-        parameters: ['boolean', 'boolean']
-    }, {
-        type: 'boolean',
-        parameters: ['null', 'null']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['string', 'value', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['number', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['boolean', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['null', 'value']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'string']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'string', 'collator']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'number']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'boolean']
-    }, {
-        type: 'boolean',
-        parameters: ['value', 'null']
-    }],
+    '==': [].concat(equalitySignatures),
+    '!=': [].concat(equalitySignatures),
+    '<': [].concat(inequalitySignatures),
```

Comment:
Why do the arrays have to be copied here instead of shared? Not that I see a problem with it, I think I'm just missing some hidden requirement.

Reply:
Eh they don't -- I don't even know why I did it

Reply:
Seems like explicitly listing out the possible combinations might be more confusing than collapsing the "formal" signature down to just (value, value), and explaining the constraints in prose. Something like...

For ==, !=:

> The comparison is strictly typed; arguments of different runtime types are considered unequal. Cases where the types are known to be different at parse time produce a parse error.

For others:

> The arguments are required to be either both strings or both numbers; if during evaluation they are not, expression evaluation produces an error. Cases where this constraint is known not to hold at parse time produce a parse error.